generators/entity/templates/server/src/main/java/package/domain/_Entity.java

```diff
@@ -58,8 +58,9 @@
 <% } -%>
 <% if (databaseType == 'sql') { -%>
 @Entity
-@Table(name = "<%= entityTableName %>")<% if (hibernateCache != 'no') { %>
-@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)<% } %><% } %><% if (databaseType == 'mongodb') { %>
+@Table(name = "<%= entityTableName %>")
+<% if (hibernateCache != 'no') { if (hibernateCache == 'infinispan') { %>@Cache(usage = CacheConcurrencyStrategy.READ_WRITE) <% } else { %>
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)<% } } %><% } %><% if (databaseType == 'mongodb') { %>
```

Comment:
It does, but the default cache mode for L2 is 'invalidation' mode. Though it is similar to replication, but it differs by,
A write operation would update the cache in the owner node and remove the stale values from the other nodes' memory (will not do an immediate replication). Benefit of this, network traffic is minimised as the invalidation messages are minimal compared to replicating the updated cache entity and the other nodes will look up the updated data in a lazy manner when needed. This mode is the recommended for L2 over replicated and distributed mode and it works only with READ_WRITE. For [more info](http://planet.jboss.org/post/hibernate_second_level_cache_improvements)